lib/xandra/frame.ex

```diff
@@ -63,5 +67,24 @@ defmodule Xandra.Frame do
   defp encode_flags(_, false), do: 0x01
   defp encode_flags(_, true), do: 0x03
 
-  defp maybe_compress_body(nil, body), do: body
+  defp flag_set?(flags, flag) do
+    (flags &&& flag) > 0
```

Comment:
Better to be strict here and use `== 1`, also I think `band(flags, flag)` read better.